### DISCUSSION TITLE
Cell phone gets very warm

### DISCUSSION THREAD

**Discussion by Tealk** [thunder-app/thunder#102] [General]
Hi,

has anyone else that when using Thunder for a long time the phone becomes very war?
Use a Pixel 6 with Graphene OS

**Comment by hjiangsu** [comment]
Hey! Yeah, this is something that I'm trying to fix/improve in the next release. To give you a brief explanation on what is happening:

There are times where you will start to notice some lag as you keep using the app for longer periods, especially with feeds containing a lot of photos. The issue right now is two-fold:
1. As more images get loaded from the feed, the images stay in your phones memory, and you'll then see that the memory usage spikes up the longer you scroll or the more images you've loaded. The longer you keep scrolling, the more lag you're likely to experience because the app uses more and more system memory. **This is currently what I'm trying to reduce for the next release**
2. Images currently only have a single resolution/size when it's retrieved from the lemmy instance. Because of this, some images that are very large (really high resolution), take more time to download and also ends up taking more system memory. Because of this, and the fact that there could be many images being downloaded at once on the background, your phone is doing a lot of processing which is what causes it to heat up a bit 

For the second point, there's no clear cut solution that I can think of to fix it as of right now. The best thing would be for the lemmy instances to provide different resolution images so that Thunder can then choose the best resolution image to display according to the phones resolution (maybe someone can chime in and provide suggestions!)

**Reply by Tealk** [reply]
Lemmy has no thumbnails for his pictures? This is of course bad, difficult to implement a solution on the client side.
Maybe as a workaround that images are only loaded when you click on them?

**Reply by hjiangsu** [reply]
I looked at their API, and there is a field for thumbnails - the thing is, from what little I've observed, it seems like the thumbnail usually is the same resolution as the actual image, so it makes very little difference

Maybe this has changed though, so thanks for bringing it back up! I'll double check to see if maybe the thumbnail logic has changed on lemmy's end

**Reply by Tealk** [reply]
no problem, if you need more tasks let me know :P
